controller/src/beerocks/master/controller_ucc_listener.cpp

```diff
@@ -68,6 +68,7 @@ std::shared_ptr<uint8_t> controller_ucc_listener::get_buffer_filled_with_cmdu()
 bool controller_ucc_listener::send_cmdu_to_destination(ieee1905_1::CmduMessageTx &cmdu_tx,
                                                        const std::string &dest_mac)
 {
+    m_database.free_certification_tx_buffer();
```

Comment:
I may be wrong, but: this should be the last shared_ptr reference to the certification buffer, so it will be freed. However, the cmdu_tx is still referring to it, just not as a shared_ptr. So cmdu_tx will be pointing to freed memory...

Anyway, I think it's the wrong thing to do. The certification buffer is filled when some specific CMDU arrives in the controller. In principle, this is more or less asynchronous to what the UCC is doing. So it's possible that the UCC first does some other dev_send_1905 before it does the dev_send_1905 which should use the certification buffer. So I think the free should be done only in the cases that we were using the certification buffer.